File: Data Structures & Algorithms/non-cyclical-number/submission-0.js
class Solution {
  /**
   * @param {number} n
   * @return {boolean}
   */
  isHappy(n) {
    const mp = new Set();
    let temp = n;
    const calSquare = (num) => {
      let square = 0;
      while (num) {
        let digit = num % 10;
        square += digit * digit;
        num = Math.floor(num / 10);
      }
      return square;
    };
    while (n != 1) {
      if (mp.has(n)) {
        return false;
      } else {
        mp.add(n);
      }
      n=calSquare(n);
    }
    return true;
  }
}

File: Data Structures & Algorithms/plus-one/submission-0.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        temp=0
        for i in digits:
            temp =temp*10 + i
        temp=temp+1
        res=[]
        while (temp):
            digit=temp%10
            temp=temp//10
            res.append(digit)
        return res[::-1]